src/sentry/models/outbox.py

```diff
@@ -121,7 +121,8 @@ def find_scheduled_shards(cls) -> Iterable[Mapping[str, Any]]:
 
     @classmethod
     def prepare_next_from_shard(cls, row: Mapping[str, Any]) -> OutboxBase | None:
-        with transaction.atomic(savepoint=False):
+        using = router.db_for_write(cls)
+        with transaction.atomic(using=using, savepoint=False):
```

Comment:
This method runs on both silos so we need to specify the connection name or the transaction will start on `default` but the queries will run on `control|region`.